tools/halide_image_io.h

```diff
@@ -1755,7 +1755,7 @@ struct halide_tiff_header {
 template<typename ElemType, int BUFFER_SIZE = 1024>
 struct ElemWriter {
     ElemWriter(FileOpener *f)
-        : f(f), next(&buf[0]), ok(true) {
```

Comment:
Here for example I think it's much clearer if the initial state for ElemWriter is written down in a single place, instead of being split between the constructor and the members.

Reply:
Interesting! I see it a different way: with this change, all of the members that are default-inited (either to a constant value or default-ctor for nontrivial types) are done so in the class declaration, leaving the classical "ctor-init-list" used only for things that are inited by an argument to the ctor itself (or by expressions that aren't otherwise constant). IMHO that is more clear.

Reply:
This particular case was what tipped me over the edge to not liking it to the point of commenting. next(&buf[0]) is not a function of the arguments, it's just the initial value for that field. I don't see it as relevant that those bits aren't a compile-time constant.

Reply:
I think this case is probably a limitation of clang-tidy being conservative (ie, being unable to prove to itself that this was a valid construct) -- `ElemType *next = &buf[0];` is perfectly valid here AFAICT (and I'll push a manual edit to this PR to fix that).

That said, this isn't an issue I want to spend any nontrivial amount of time on; I'll just open a new PR with the new clang-tidy behavior disabled and see how that looks. (The main desire here is just to keep our tooling relatively new, not to enforce arbitrary new checks.)